raster-foundry/raster-foundry
# Upgrade RF client to Python3

## Overview

This commit includes the minimal set of changes required to run `rf`
commands in python 3. Additionally, a few commands were removed that
are no longer necessary:
 - find-aoi-projects
 - reprocessing landsat and sentinel imagery
 - updating aoi projects

The MODIS/CMR processing wasn't extensively tested since that is being
removed in an upcoming PR.

### Checklist


### Demo

Successful Landsat 8 Ingest in Application
![landsat8-ingest-app](https://user-images.githubusercontent.com/898060/69078206-59725200-0a05-11ea-90b9-fb93ad98a646.png)

Successful Landsat 8 ingest CLI
![landsat8-ingest](https://user-images.githubusercontent.com/898060/69078235-6e4ee580-0a05-11ea-95bd-5fee2b9e4aff.png)

Successful User Upload -> COG
![process-upload](https://user-images.githubusercontent.com/898060/69078273-81fa4c00-0a05-11ea-96c6-1ce0d14dd7f0.png)

Successful Export Application
![image](https://user-images.githubusercontent.com/898060/69078326-a22a0b00-0a05-11ea-883e-3bd0e62fc8c1.png)

Successful Export QGIS
![image](https://user-images.githubusercontent.com/898060/69078383-c4bc2400-0a05-11ea-9fef-327287cb4c99.png)

Successful Ingest Sentinel 2 (Terminal)
![image](https://user-images.githubusercontent.com/898060/69078413-da314e00-0a05-11ea-98e0-d9a6cc4dfd89.png)

## Testing Instructions

Below are testing instructions - I tried to include screenshots for everything, but I went through these verbatim to test the following:
 - Exports
 - Ingests (Sentinel 2 & Landsat 8)
 - User Upload

I would suggest maybe not testing this since it takes a while and exercising the commands should be enough to reveal any issues with the upgrade since most of the python code is glue between command line utilities and scala code.

- Update batch container (`docker-compose build batch`), server assemblies
- Get auth token and export it: `export JWT_AUTH_TOKEN=<token>`
- Create an export:
```bash
echo '{"projectId":"94bb99c1-c8da-4321-a03b-fce884009a90","exportStatus":"NOTEXPORTED","exportType":"S3","visibility":"PRIVATE","exportOptions":{"resolution":11,"crop":false,"raw":false,"mask":{"type":"MultiPolygon","coordinates":[[[[-107.529736,36.226074],[-107.531608,36.187282],[-107.454849,36.16813],[-107.447985,36.212474],[-107.529736,36.226074]]]]},"bands":[0,1,2,3]},"projectLayerId":"32a99ebc-83f0-4178-8c50-c70e66e343ba"}' | http --auth-type=jwt :9100/api/exports
```
- Grab the ID and run an export from RF
```bash
./scripts/console batch "rf export <export-id>"
```
- Ingest Landsat 8 Scene
```bash
./scripts/console batch "rf ingest-scene b0557dc2-1766-43db-b4be-ddd914e6566b"
```
- Ingest Sentinel 2 Scene
```bash
./scripts/console batch "rf ingest-scene e0b3bef2-8cfe-4026-822a-d4bf85337e58"
```
- Upload tif and verify that upload processing still works (`s3://rasterfoundry-development-data-us-east-1/phila-sample.tif`)
```bash
echo '{"files":["s3://rasterfoundry-development-data-us-east-1/phila-sample.tif"],"datasource":"b4ca6661-27fa-497e-a85a-2a5632d5fe7c","fileType":"GEOTIFF","uploadStatus":"UPLOADED","visibility":"PRIVATE","metadata":{"acquisitionDate":"2019-11-18T16:19:17.630Z","cloudCover":0},"uploadType":"LOCAL","projectId":"410d4975-0eed-40c5-aa5a-511277507191"}' | http --auth-type=jwt :9100/api/uploads
./scripts/console batch "rf process-upload <upload-id>"
```

Closes #5243
Closes #5242
